src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -203,7 +207,7 @@ protected void doHealthCheck(Health.Builder builder) {
 
 		long startTime = System.currentTimeMillis();
 		while (true) {
-			String receivedMessage = cache.getIfPresent(expectedMessage);
+			String receivedMessage = cache.getIfPresent(consumerGroupId);
```

Comment:
Since the `consumerGroupId` doesn't change at runtime, the cache no longer makes sense, because there will only be at most one entry. I think you can't use `consumerGroupId` as a key.